modules/core/src/main/java/org/apache/synapse/deployers/ProxyServiceDeployer.java

```diff
@@ -79,32 +80,35 @@ public String deploySynapseArtifact(OMElement artifactConfig, String filePath,
                 }
 
                 getSynapseConfiguration().addProxyService(proxy.getName(), proxy);
-                AxisService axisService = proxy.buildAxisService(getSynapseConfiguration(),
-                                                  getSynapseConfiguration().getAxisConfiguration());
+                try {
+                    AxisService axisService = proxy.buildAxisService(getSynapseConfiguration(),
+                                                                     getSynapseConfiguration().getAxisConfiguration());
+
+                    if (axisService == null) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Skipping proxy Startup for ProxyService : " + proxy.getName());
+                        }
+                        return proxy.getName();
+                    }
 
-                if (axisService == null) {
                     if (log.isDebugEnabled()) {
-                        log.debug("Skipping proxy Startup for ProxyService : " + proxy.getName());
+                        log.debug("Started the ProxyService : " + proxy.getName());
+                        log.debug("ProxyService Deployment from file : " + filePath + " : Completed");
                     }
-                    return proxy.getName();
-                }
-
-                if (log.isDebugEnabled()) {
-                    log.debug("Started the ProxyService : " + proxy.getName());
-                    log.debug("ProxyService Deployment from file : " + filePath + " : Completed");
-                }
-
-                log.info("ProxyService named '" + proxy.getName()
-                        + "' has been deployed from file : " + filePath);
 
-                if (!proxy.isStartOnLoad() || !axisService.isActive()) {
-                    proxy.stop(getSynapseConfiguration());
                     log.info("ProxyService named '" + proxy.getName()
-                             + "' has been stopped as startOnLoad parameter is set to false");
-                }
-
-                return proxy.getName();
+                             + "' has been deployed from file : " + filePath);
 
+                    if (!proxy.isStartOnLoad() || !axisService.isActive()) {
+                        proxy.stop(getSynapseConfiguration());
+                        log.info("ProxyService named '" + proxy.getName()
+                                 + "' has been stopped as startOnLoad parameter is set to false");
+                    }
+                    return proxy.getName();
+                } catch (SynapseException e) {
+                    getSynapseConfiguration().removeProxyService(proxy.getName());
+                    throw e;
```

Comment:
Shall we use handleSynapseArtifactDeploymentError method here ?

Reply:
Caught in https://github.com/wso2/wso2-synapse/pull/1273/files#diff-771ac85f9b9e01d365ddef0c512dfdeaR124